neo_batterylevelshutdown/neo_batterylevelshutdown.py

```diff
@@ -69,39 +74,36 @@ def blink_LEDxTimes(pinNum, times):
 
 def readPin(pinNum):
     """Read the value from some input pin"""
+    logging.info("Reading pin {}".format(pinNum))
     try:
-        with open("/sys/class/gpio/gpio{}/value".format(pinNum)) as pin:
-            return pin.read(1) == 1
+        with open("/sys/class/gpio/gpio{}/value".format(pinNum), 'r') as pin:
+            return str(pin.read(1)) == "1"
     except OSError:
         logging.warn("Error reading from pin {}".format(pinNum))
-
     return -1
 
-
 def entryPoint():
+#if __name__ == "__main__":
+    #logging.basicConfig(filename='/root/output.log', filemode='w', level=logging.DEBUG)
     logging.info("Intializing Pins")
-    logging.debug("Pin: LED")
-    setup_gpio_pin(PIN_LED, "out")
-    logging.debug("Pin: PG6 3.0V")
-    setup_gpio_pin(PIN_VOLT_3_0, "in")
-    logging.debug("Pin: PG7 3.2V")
-    setup_gpio_pin(PIN_VOLT_3_2, "in")
-    logging.debug("Pin: PG8 3.4V")
-    setup_gpio_pin(PIN_VOLT_3_4, "in")
-    logging.debug("Pin: PG9 3.6V")
-    setup_gpio_pin(PIN_VOLT_3_6, "in")
+    setup_gpio_pin(PIN_LED, "out", 0)
+    setup_gpio_pin(PIN_VOLT_3_0, "in", 1) # set the pulldown resistor
```

Comment:
This seems to be the secret sauce to avoid us shutting down on devices that don’t have a HAT. Please add a comment to explain why we need to do this. Perhaps something like:

```
# On devices that do not have a HAT, PIN_VOLT_3_0 will be low and we will
#  immediately shut down. We set it high by default to avoid this situation
#  knowing that the HAT will set low when the voltage drops below the
#  3.0v threshold
```

(I’m not sure that’s actually correct, but it’s the sort of info that we should add to explain the logic)

Also, what happens if we start the board and the battery level is such that the HAT would want to trigger a shutdown? Does this setting of the pulldown resistor to up cause the device to not shutdown? i.e. does the HAT do a single transition once the voltage drops below a certain value, or does it repeatedly set it?